index.js

```diff
@@ -840,58 +811,33 @@ const ContainerService = {
   },
 
   openTab(args) {
-    return new Promise(resolve => {
-      if ("window" in args && args.window) {
-        resolve(args.window);
-      } else {
-        this._recentBrowserWindow().then(browserWin => {
-          resolve(browserWin);
-        }).catch(() => {});
-      }
-    }).then(browserWin => {
-      const userContextId = ("userContextId" in args) ? args.userContextId : 0;
-      const source = ("source" in args) ? args.source : null;
-      const nofocus = ("nofocus" in args) ? args.nofocus : false;
-
-      // Only send telemetry for tabs opened by UI - i.e., not via showTabs
-      if (source && userContextId) {
-        this.sendTelemetryPayload({
-          "event": "open-tab",
-          "eventSource": source,
-          "userContextId": userContextId,
-          "clickedContainerTabCount": this._containerTabCount(userContextId)
-        });
-      }
+    return this.triggerBackgroundCallback(args, "open-tab");
+  },
 
-      let promise;
-      if (userContextId) {
-        promise = this.showTabs(args);
-      } else {
-        promise = Promise.resolve(null);
-      }
+  // Identities management
+  queryIdentitiesState() {
```

Comment:
The follow up to this can move out the state tracking into the background page after hide and sort is moved there too.